### ISSUE TITLE
Error on salt-ssh call utf8 encoding issue "[ERROR] JSON Render failed for"

### ISSUE THREAD

**Issue by cebe** [saltstack/salt#38795]
### Description of Issue/Question

One of my states fails with the error `[ERROR] JSON Render failed for` when I try to apply it. The state seems to be applied in the background but the result output is broken. See below.

Looks similar to #37906, #16651

### Setup

I can reproduce this issue with the following state file located at `salt://php/wtf.sls`:

```yaml
# clone git repo for php build tools
make_php_repo:
  git.latest:
    - name: https://github.com/cebe/make-php.git
    - rev: 0f0817f4f1202d2e47ed771363fe4c6248217b11 # Jan 17, 2017
    - target: /tmp/test

# update gpg keyring
/tmp/test/make-php-keyring.sh:
  cmd.run:
    - require:
      - git: make_php_repo
```

### Steps to Reproduce Issue

applying it like this:

```
salt-ssh 'dvm2*' state.sls php.wtf
```

output:

```
[ERROR   ] JSON Render failed for: 
Traceback (most recent call last):
  File "/var/tmp/.root_8cdb53_salt/salt-call", line 15, in <module>
    salt_call()
  File "/var/tmp/.root_8cdb53_salt/py2/salt/scripts.py", line 352, in salt_call
    client.run()
  File "/var/tmp/.root_8cdb53_salt/py2/salt/cli/call.py", line 58, in run
    caller.run()
  File "/var/tmp/.root_8cdb53_salt/py2/salt/cli/caller.py", line 149, in run
    self.opts)
  File "/var/tmp/.root_8cdb53_salt/py2/salt/output/__init__.py", line 86, in display_output
    display_data = try_printout(data, out, opts)
  File "/var/tmp/.root_8cdb53_salt/py2/salt/output/__init__.py", line 39, in try_printout
    return get_printout(out, opts)(data).rstrip()
  File "/var/tmp/.root_8cdb53_salt/py2/salt/output/json_out.py", line 57, in output
    return json.dumps(data, default=repr, indent=4)
  File "/usr/lib/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/lib/python2.7/json/encoder.py", line 434, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 390, in _iterencode_dict
    yield _encoder(value)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xfc in position 792: invalid start byte
[ERROR   ] No JSON object could be decoded
dvm2.spinitron.local:
```

### Versions Report

```
$ salt-ssh --versions-report
Salt Version:
           Salt: 2016.3.3
 
Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.0
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: Not Installed
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: Not Installed
 
System Versions:
           dist: debian 8.6 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.6 
 
```

**Comment by cebe** [comment]
Seems to be fixed by piping stderr of the script to `/dev/null`.

however I do not see any problematic character in the output when I run the script myself.

**Comment by Ch3LL** [comment]
@cebe what if you use cmd.script instead of cmd.run any luck?

**Comment by cebe** [comment]
@Ch3LL thanks for the quick feedback! Sadly that does not change anything, ran it as:

```yaml
/tmp/test/make-php-keyring.sh:
  cmd.script:
    - require:
      - git: make_php_repo
```

As stated above I can work around this by piping stderr to /dev/null, quite sure I could use a file if I needed, so that solves it for my case. I still think this should be fixed so it does not crash but display a warning instead, so the result of other states would still be shown. There are 2 options imo:

- fix it to strip invalid characters from the output and display the rest. Could be combined with a warning about these characters
- do not display the command output and place a warning instead if this exception is thrown. I.e. catch the exception and convert it to a warning.

**Comment by Ch3LL** [comment]
@cebe looks like i'm able to replicate this using your same state file on 2016.11.1 and agreed we should be parsing the return data here better so we don't get a unicode error. We will need to get this fixed up thanks.

**Comment by Ch3LL** [comment]
Also forgot to mention that this state does work with running `salt-call`. So its just salt-ssh that is having the issue here. Thanks

**Comment by cebe** [comment]
This seems to be fixed in "salt-ssh 2018.3.2"

```
# salt-ssh --version
salt-ssh 2018.3.2 (Oxygen)
# salt-ssh -i '*' test.ping
test:
    True
# salt-ssh -i '*' state.sls php.wtf
test:
----------
          ID: /tmp/test/make-php-keyring.sh
    Function: cmd.run
      Result: True
     Comment: Command "/tmp/test/make-php-keyring.sh" run
     Started: 11:38:39.919201
    Duration: 11206.929 ms
     Changes:   
              ----------
              pid:
                  6982
              retcode:
                  0
              stderr:
                  gpg: key F50ABC807BD5DCD0: "Davey Shafik <me@daveyshafik.com>" not changed
                  gpg: key BCAA30EA9C0D5763: "Anatol Belski <ab@php.net>" not changed
                  gpg: key 7DEC4E69FC9C83D7: "Johannes Schl�ter <johannes@schlueters.de>" not changed
                  gpg: key 2F7956BC5DA04B5D: "Stanislav Malyshev (PHP key) <smalyshev@gmail.com>" not changed
                  gpg: key C13C70B87267B52D: "David Soria Parra <dsp@php.net>" not changed
                  gpg: key FE857D9A90D90EC1: "Julien Pauli <jpauli@php.net>" not changed
                  gpg: key C2BF0BC433CFC8B3: "Ferenc Kovacs <tyrael@php.net>" not changed
                  gpg: Total number processed: 7
                  gpg:              unchanged: 7
              stdout:

Summary for test
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
Total run time:  12.973 s
```
